### ISSUE TITLE
Doesn't work well with low probability values

### ISSUE THREAD

**Issue by harishkashyap** [pgmpy/pgmpy#952]
### Subject of the issue
It doesnt work well with low probability values. For this repository to be used for applications such as fraud modeling, then we have to work with low values. Any value beyond 1e-5 is treated as 0.

### Your environment
* pgmpy version: latest
* Python version : python 3
* Operating System : Mac OS

### Steps to reproduce
I will provide the example code.

### Expected behaviour
For Marginal probaiblities where low values need to be received, a completely erroneous value gets generated. Best is to use log values everywhere for this package.

### Actual behaviour
You get 1.0 or 0.0. This is a serious error in output probabilities.

**Comment by ankurankan** [comment]
@harishkashyap Could you please share your code so that we can reproduce this? 

This seems like a really difficult thing to solve, since lower values would always go towards 0 when multiplied with other small values. Only possible way I can think of dealing with this would be to scale all the probability values before doing any operations on it. @khalibartan @yashu-seth Thoughts ?

**Comment by harishkashyap** [comment]
@ankurankan firstly, great job on this! I am using this for a workshop since no other library seems close enough. However, it is critical to get this working for low probability values as if it is rounded off to 0 then we run the risk of other values. Esp, in fraud modeling you would have 1e-7/1e-9 where the values though are low to begin with will end up making sense later. I would suggest using log and not scaling. Scaling them will again introduce approximate values which can be riskier.

**Comment by ankurankan** [comment]
@harishkashyap Yeah, I was suggesting log scaling as well. I checked the PR that you created and the values aren't that small in the example there. Could you please share the code where the values are getting rounded ? Because that shouldn't actually happen, since our `DiscreteFactor` and `TabularCPD` classes use `float64`, numpy should be able to support values till `2.22044604925e-16`, which is quite small. By any chance are you using a 32 bit system ?

**Comment by harishkashyap** [comment]
@ankurankan I updated the code. Let me know if I am doing something wrong. I am using a Macbook Pro latest version
